### PR TITLE
Spec creating users

### PR DESCRIPTION
**Pull request by oponder**
I want a way to create users in the new /v4/ paradigm. This will let me add validations and sanitization that are missing currently at the v1 endpoint.

It also lets us get closer to a point where we can remove all legacy from api, which is something I really want to do.

I know we intend to replace user management entirely in the short/medium term, but I don't want to have any more tech debt and confusion caused by things we know but are not fixing.

### CODE REVIEW

File: spec.yaml [hunk 1]
```diff
@@ -109,6 +109,58 @@ paths:
...
+              }
+        "400":
+          description: User already exists
+          schema:
+            $ref: "definitions.yaml#/definitions/V4GenericResponse"
+          examples:
+            application/json:
+              {
+                "code": "RESOURCE_ALREADY_EXISTS",
+                "message": "The user could not be created. (invalid input: email 'bob@example.com' already exists.)"
```
Comment (xh3b4sd) [inline spec.yaml]: The full stop at the end of the error message in the braces is not right IMO.

File: definitions.yaml [hunk 2]
```diff
@@ -226,3 +226,16 @@ definitions:
...
+
+  V4CreateUserRequest:
+    type: object
+    required:
+      - password
+    description: Request model for creating a new user
+    properties:
+      password:
+        type: string
+        description: A Base64 encoded password
```
Comment (marians) [inline definitions.yaml]: Not sure about the base64 encoding, as it's not a solid standard. See also my comments on https://github.com/giantswarm/api-spec/pull/49

**Comment by oponder** [comment]
Got any strong feelings about the base64 encoding? I am happy to adjust that, but I finally have an implementation for this now.

File: definitions.yaml (hunk 2 above)
Comment (puja108) [reply definitions.yaml]: not sure, but it seems to work quite well in k8s

**Comment by marians** [comment]
No more strong feelings. The fact that it works well for kubernetes secrets is convincing.

**Pull request merged**